### ISSUE TITLE
FxR crashes when playing YouTube videos

### ISSUE THREAD

**Issue by kaipinglee** [MozillaReality/FirefoxReality#2381]
## Configuration

Firefox Reality version: v5
Firefox Reality build ID: 

Hardware: Vive Focus plus

## Steps to Reproduce

0. HTC customer service department reported 10+ consumer issues regarding YouTube.
1. Use FxR to play Youtube videos.
2. Sometime FxR screen stop responding during YouTube video playing.
3. FxR prompt "We had an issue and crashed" dialog and request user to send data.

![2019-11-27 17_32_35-htceasap1 htc com tw_its_servlet_SqaLogDownloadServlet_fileName=UITS_324692_FA94](https://user-images.githubusercontent.com/31957856/69711583-487bbc00-113c-11ea-8941-fbe245ec022d.png)

![2019-11-27 17_33_21-htceasap1 htc com tw_its_servlet_SqaLogDownloadServlet_fileName=UITS_324693_FA94](https://user-images.githubusercontent.com/31957856/69711612-503b6080-113c-11ea-8450-0ba1f83d23c9.png)

## Current Behavior

FxR crashes when playing YouTube videos. 

## Expected Behavior

1. FxR can play YouTube videos normally.
2. Change the "We had an issue and crashed." string if it is YouTube server-side error and auto recover from the exception.

## Error Logs and Stack Traces

Refer to below logs.
[https://drive.google.com/open?id=1aD_ZFE-PDToaktfCPouMYr9GG-le71i-](url)
[https://drive.google.com/open?id=1PKUIhLbapRvvu8aps5r4xS1870jJCgVD](url)
[https://drive.google.com/open?id=10vF4CzWO9hjwFFNr4ctIQzjyyK0YlAsw](url)
<details open>

```
```

</details>

**Comment by MortimerGoro** [comment]
This seems to be the crash:

```
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: >>> REPORTING UNCAUGHT EXCEPTION FROM THREAD 152 ("FinalizerWatchdogDaemon")
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: java.util.concurrent.TimeoutException: android.view.ThreadedRenderer.finalize() timed out after 15 seconds
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: 	at android.view.ThreadedRenderer.nDeleteProxy(Native Method)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: 	at android.view.ThreadedRenderer.finalize(ThreadedRenderer.java:919)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: 	at java.lang.Daemons$FinalizerDaemon.doFinalize(Daemons.java:233)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: 	at java.lang.Daemons$FinalizerDaemon.run(Daemons.java:220)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: 	at java.lang.Thread.run(Thread.java:761)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler: Main thread (1) stack:
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.view.ThreadedRenderer.nDestroyHardwareResources(Native Method)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.view.ThreadedRenderer.destroyHardwareResources(ThreadedRenderer.java:532)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.view.ViewRootImpl.destroyHardwareResources(ViewRootImpl.java:827)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.view.WindowManagerGlobal.trimMemory(WindowManagerGlobal.java:498)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.app.ActivityThread.handleTrimMemory(ActivityThread.java:5392)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.app.ActivityThread$H.handleMessage(ActivityThread.java:1738)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.os.Handler.dispatchMessage(Handler.java:102)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.os.Looper.loop(Looper.java:173)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     android.app.ActivityThread.main(ActivityThread.java:6608)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     java.lang.reflect.Method.invoke(Native Method)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:972)
11-08 12:15:09.737  8213  8222 E GeckoCrashHandler:     com.android.internal.os.ZygoteInit.main(ZygoteInit.java:852)
```

**Comment by MortimerGoro** [comment]
The crash reports are from FxR v5 and it seems related to a Android hard acceleration bug (related to #1959=
@Softvision-RemusDranca or @Softvision-GeluHaiduc Can you check test this one on a Focus device? It may have been already fixed by #2184

**Comment by Softvision-GeluHaiduc** [comment]
I managed to partially reproduce the issue using Firefox Reality version 7 RC3 (7b66fb14) installed on HTC Vive Focus Plus.
After more than 1 hour of stressing the app navigating Youtube and quitting and resuming the app at 1 video the Black screen with the error message appeared, the logcat file can be found [here](https://drive.google.com/open?id=1_1GxfULPQvw2tk_CYP8z09miKr-IEGcr) and the screen recording [here](https://drive.google.com/open?id=1_rPIzoZ5cUx8hle_Iit3Ktt0AibWV_zC).
After some more stressing in which I created tabs and windows and again quitting and resuming the app crashed, the logcat file is [here](https://drive.google.com/open?id=1nMt0Z6kDQ-cO-he5YT3Nlln6L7hgTpA1) and the crash report [here](https://crash-stats.mozilla.com/report/index/0451006b-4be4-47c7-97e9-749950191127).

**Comment by MortimerGoro** [comment]
Thanks for the info @Softvision-GeluHaiduc!

I created a separate issue for the crash #2383.  It's different from the one that @kaipinglee reported, which seems fixed with the #2184

**Comment by MortimerGoro** [comment]
Fixed in #2482

**Comment by Softvision-GeluHaiduc** [comment]
I have verified this issue with the latest Firefox Reality version 8-40101959 (3c2a71e2) installed on the HTC Vive Focus and HTC Vive Focus Plus headsets.
From what we tested the issue is no longer reproducible at this time, but keep in mind that it is an intermittent issue. If it reappears in the future we will file a new issue for that.
